experiment/oss_fuzz_checkout.py

```diff
@@ -375,23 +375,58 @@ def prepare_build(project_name, sanitizer, generated_project):
     shutil.copy(original_dockerfile, dockerfile_to_use)
 
 
-def _image_exists(image_name: str) -> bool:
-  """Checks if the given |image_name| exits."""
+def _image_exists_locally(image_name: str, project_name: str) -> bool:
+  """Checks if the given |image_name| exits locally."""
   try:
     all_images = sp.run(['docker', 'images', '--format', '{{.Repository}}'],
                         stdout=sp.PIPE,
                         text=True,
                         check=True).stdout.splitlines()
+    if image_name in all_images:
+      logger.info('Will use local cached images of %s: %s', project_name,
+                  image_name)
+      return True
   except sp.CalledProcessError:
-    logger.info('Unable to list all docker images')
+    logger.warning('Unable to use local cached image of %s: %s', project_name,
+                   image_name)
+  return False
+
+
+def _image_exists_online(image_name: str, project_name: str) -> bool:
```

Comment:
Can you re-use existing functionality in this file? see e.g. https://github.com/google/oss-fuzz-gen/blob/0d80846b50a5d2dbb98e9dc154edb5ab34d06da5/experiment/oss_fuzz_checkout.py#L189C5-L189C34

Reply:
Done. I also thought about reusing other functions, but they appear to be coupled with cached build scripts etc.
I want to avoid adding params to them for my case, but please let me know if you think that's better.